tests/Doctrine/Tests/DBAL/ConnectionTest.php

```diff
@@ -198,7 +198,10 @@ public function testDriverExceptionIsWrapped($method)
         $connection->$method('MUUHAAAAHAAAA');
     }
 
-    public function getQueryMethods()
+    /**
+     * @return array<int, array<int, mixed>>
```

Comment:
Should it be `array<int, array<int, string>>`? I'd usually try to get it as close as possible to what the provider is returning but also understand it can be overkill sometimes so I am ok with it being `array<int, array<int, mixed>>` for all providers.

Reply:
The inner type (`array<int, mixed>`) represents a function arguments. Even if we pass them as an array, they are not. They are a tuple where each element has its own independent type. Even if they all happen to be of a certain type at a certain place (e.g. all strings), it doesn't mean they have to be. If tomorrow I add one more argument to the test method and reflect it in the data provider, it shouldn't require changing the types.